examples/placeholder/linux/InteractiveServer.cpp

```diff
@@ -32,6 +34,117 @@ constexpr const char * kAttributeWriteKey           = "writeAttribute";
 constexpr const char * kAttributeReadKey            = "readAttribute";
 constexpr const char * kCommandIdKey                = "commandId";
 constexpr const char * kWaitForCommissioningCommand = "WaitForCommissioning";
+constexpr const char * kCategoryError               = "Error";
+constexpr const char * kCategoryProgress            = "Info";
+constexpr const char * kCategoryDetail              = "Debug";
+
+struct InteractiveServerResultLog
+{
+    std::string module;
+    std::string message;
+    std::string messageType;
+};
+
+struct InteractiveServerResult
+{
+    bool mEnabled = false;
+    std::vector<std::string> mResults;
+    std::vector<InteractiveServerResultLog> mLogs;
+
+    void Setup() { mEnabled = true; }
+
+    void Reset()
+    {
+        mEnabled = false;
+        mResults.clear();
+        mLogs.clear();
+    }
+
+    void MaybeAddLog(const char * module, uint8_t category, const char * base64Message)
+    {
+        VerifyOrReturn(mEnabled);
+
+        const char * messageType = nullptr;
+        switch (category)
+        {
+        case chip::Logging::kLogCategory_Error:
+            messageType = kCategoryError;
+            break;
+        case chip::Logging::kLogCategory_Progress:
+            messageType = kCategoryProgress;
+            break;
+        case chip::Logging::kLogCategory_Detail:
+            messageType = kCategoryDetail;
+            break;
+        }
+
+        mLogs.push_back(InteractiveServerResultLog({ module, base64Message, messageType }));
+    }
+
+    void MaybeAddResult(const char * result)
+    {
+        VerifyOrReturn(mEnabled);
+        mResults.push_back(result);
+    }
+
+    std::string AsJsonString() const
+    {
+        std::string resultsStr;
+        if (mResults.size())
+        {
+            for (const auto & result : mResults)
+            {
+                resultsStr = resultsStr + result + ",";
+            }
+
+            // Remove last comma.
+            resultsStr.pop_back();
+        }
+
+        std::string logsStr;
+        if (mLogs.size())
+        {
+            for (const auto & log : mLogs)
+            {
+                logsStr = logsStr + "{";
+                logsStr = logsStr + "  \"module\": \"" + log.module + "\",";
+                logsStr = logsStr + "  \"category\": \"" + log.messageType + "\",";
+                logsStr = logsStr + "  \"message\": \"" + log.message + "\"";
```

Comment:
Won't this break if the message contains `"` characters?  I guess the base64 encoding makes that safe?  If so, document?